### PR TITLE
docs: document use of {{ .Data }} field in email template

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

This PR documents the `{{ .Data }}` field for which allows developer to include data from the `User.Metadata` field. 

Initially, this was also intended to document the `{{ .SendingTo }}` field but that has been removed  and will be addressed in a separate PR.

### CODE REVIEW

File: studio/stores/authConfig/schema/AuthProviders/AuthTemplatesValidation.tsx [hunk 1]
```diff
@@ -127,6 +127,7 @@ export const EMAIL_CHANGE: FormSchema = {
 - \`{{ .Email }}\` : The original user's email address
 - \`{{ .NewEmail }}\` : The user's new email address
 - \`{{ .Data }}\` : The user's \`user_metadata\`
+- \`{{ .SendingTo }}\: The recipient of the email. Relevant for Secure Email Change.`
```
Comment (kangmingtay) [inline studio/stores/authConfig/schema/AuthProviders/AuthTemplatesValidation.tsx:130]: maybe a link to the docs explaining why it's relevant when the "Secure Email Change" setting is enabled would be more useful here

File: apps/docs/pages/guides/auth/auth-email-templates.mdx [hunk 2]
```diff
@@ -71,6 +72,39 @@ const { data: { session }, error } = supabase.auth.verifyOtp({ token_hash, type
...
+
+Send two distinct emails: one customized for the new email account and another for the email the user is currently using.
+```
+{{ if eq .Email .SendingTo }}
+<insert content which will be received by current email>
+{{ else }}
+<insert content which will be received by new email>
+{{end}}
+
+```
```
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/auth-email-templates.mdx:104]: can we give an example of why this is useful (i.e. for the token hash validation)?

**Review by kangmingtay — CHANGES_REQUESTED** [review]

File: apps/docs/pages/guides/auth/auth-email-templates.mdx [hunk 3]
```diff
@@ -73,6 +74,33 @@ const { data: { session }, error } = await supabase.auth.verifyOtp({ token_hash,
 
 ```
 
+## Customization
+
+Supabase Auth makes use of [Go Templates](https://pkg.go.dev/text/template). This means it is possible to conditionally render information based on template properties.You may wish to checkout this [guide by Hugo](https://gohugo.io/templates/introduction/) for a guide on the templating language.
+
+Let's demonstrate some ways of using the templating system below:
```
Comment (J0) [inline apps/docs/pages/guides/auth/auth-email-templates.mdx:81]:
```suggestion
```

**Review by kangmingtay — APPROVED** [review]

File: apps/docs/pages/guides/auth/auth-email-templates.mdx [hunk 4]
```diff
@@ -73,6 +74,31 @@ const { data: { session }, error } = await supabase.auth.verifyOtp({ token_hash,
...
+  <p>Best Regards,<br>
+Billy Team</p>
+{{ else if eq .Data.Domain "https://www.earlyaccess.trial.com" }}
+<h1>Welcome to Our Database Service!</h1>
+  <p>Dear Developer,</p>
+  <p>Welcome Billy, the scalable developer platform!</p>
+  <p> As an early access member, you have access to select features like Point To Space Restoration.</p>
+  <p>Best Regards,<br>
+Billy Team</p>
+{{end }}
```
Comment (J0) [inline apps/docs/pages/guides/auth/auth-email-templates.mdx:99]:
```suggestion
{{ end }}
```

**Pull request merged**